tsdb/functions.go

```diff
@@ -1157,24 +1161,49 @@ type PositionPoint struct {
 }
 
 type topMapOut struct {
-	positionOut
+	*positionOut
+}
+
+func (t *topMapOut) Len() int      { return len(t.points) }
+func (t *topMapOut) Swap(i, j int) { t.points[i], t.points[j] = t.points[j], t.points[i] }
+func (t *topMapOut) Less(i, j int) bool {
+	return t.positionPointLess(&t.points[i], &t.points[j])
 }
 
-func (t topMapOut) Len() int      { return len(t.points) }
-func (t topMapOut) Swap(i, j int) { t.points[i], t.points[j] = t.points[j], t.points[i] }
-func (t topMapOut) Less(i, j int) bool {
+func (t *topMapOut) positionPointLess(pa, pb *PositionPoint) bool {
 	// old C trick makes this code easier to read. Imagine
```

Comment:
Back in C days, strcmp would return -1 if `A < B`, 0 if `A == B`, 1 if `A > B`. I follow this convention throughout for this code.

Now, when you are asking for a predicate, say `a < b`, the way you convert that into strcmp is `strcmp(a, b) < 0`. You move the < outside strcmp and compare against zero. Same works for `>=`, `==`, `!=` and so forth.